### PR TITLE
feat(indexer): explicity state all contracts of interest for L1/L2 prior to launch

### PR DESCRIPTION
**Pull request by hamdiallam**
Even though the primary use case is bridging, we list out all optimism contracts
that emit events for the indexer to process.

Until we have backfill support, it's important to hardcode these lists. Any addition
of a new contract will require resync until the we can dynamically backfill new contracts.
For this reason, we explicitly configure `L1Contracts` and `L2Contracts` that must be fully
specified, otherwise error out

### CODE REVIEW

**Comment by hamdiallam** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7206** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7206" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7206?utm_source=stack-comment).

File: indexer/config/presets.go [hunk 1]
```diff
@@ -4,73 +4,109 @@ import (
...
-		L1Contracts: L1Contracts{
-			OptimismPortalProxy:             common.HexToAddress("0xbEb5Fc579115071764c7423A4f12eDde41f106Ed"),
-			L2OutputOracleProxy:             common.HexToAddress("0xdfe97868233d1aa22e815a266982f2cf17685a27"),
-			L1CrossDomainMessengerProxy:     common.HexToAddress("0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1"),
-			L1StandardBridgeProxy:           common.HexToAddress("0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1"),
-			LegacyCanonicalTransactionChain: common.HexToAddress("0x5e4e65926ba27467555eb562121fac00d24e9dd2"),
+		Name: "Optimism Mainnet",
+		ChainConfig: ChainConfig{
+			L1Contracts: L1Contracts{
+				AddressManager:                  common.HexToAddress("0xdE1FCfB0851916CA5101820A69b13a4E276bd81F"),
```
Comment (hamdiallam) [inline indexer/config/presets.go]: holding open just until I can re-check these

cc @roninjin10 would appreciate a second eye on all these contract addresses in this file

**Review by roninjin10 — APPROVED** [review]

File: indexer/config/presets.go (hunk 1 above)
Comment (roninjin10) [reply indexer/config/presets.go]: If you want piece of mind you can validate them in a script against these addresses here. These addresses are completely generated from our artifacts. https://github.com/ethereum-optimism/optimism/pull/6301/files#diff-b820215d5bd670fe39c050ac07fa2c98e3150b080033fe34b290981387da9207R126

Comment (hamdiallam) [reply indexer/config/presets.go]: triple checked everything

**Pull request merged**